lib/p2p/packets/types/HelloPacket.ts

```diff
@@ -0,0 +1,23 @@
+import Packet, { MessageType } from '../Packet';
+import PacketType from '../PacketType';
+
+type HelloPacketBody = {
+  version: string;
+  nodeKey: string;
+  listenPort: number;
+  pairs: string[];
+};
+
+class HelloPacket extends Packet {
+  public type: PacketType = PacketType.HELLO;
+  public messageType: MessageType = MessageType.UNILATERAL;
+  public body!: HelloPacketBody;
+
+  public init(args: HelloPacketBody): HelloPacket {
```

Comment:
This init pattern seems unnecessary and somewhat counterintuitive. Why not just make this a constructor method that accepts a `HelloPacketBody` param. I don't see the difference between `new HelloPacket(body)` and `new HelloPacket().init(body)`.

Reply:
It's mainly to avoid constructor overloading (which isn't so straight-forward in TS as I thought), and to be aligned with the non-static approach for the `toRaw/fromRaw`, which its reasons I specified in the other comment. 
Declaring factory methods on an instance is indeed quite counterintuitive, but isn't so uncommon, due to the flexibility it gives.